aten/src/ATen/core/Tensor.h

```diff
@@ -280,12 +280,13 @@ class CAFFE2_API Tensor {
   /// TensorOptions.h.
   TensorOptions options() const;
 
-  void* data_ptr() const {
-    return this->unsafeGetTensorImpl()->data();
-  }
+  template <typename T = void>
```

Comment:
We provide `void` as the default template type here, so that the original `tensor.data_ptr()` call will still return a `void*` pointer.